src/types.ts

```diff
@@ -28,7 +28,7 @@ export type UmzugOptions<Ctx extends {} = Record<string, unknown>> = {
 		 * Should return an array of [filepath, content] pairs. Usually, only one pair is needed, but to put `down` migrations in a separate
 		 * file, more than one can be returned.
 		 */
-		template?: (filepath: string) => Array<[string, string]>;
+		template?: (filepath: string) => Array<[string, string]> | Promise<Array<[string, string]>>;
```

Comment:
We have a helper for this

```suggestion
		template?: (filepath: string) => Promisable<Array<[string, string]>>;
```